backend/hubs/views/hub_views.py

```diff
@@ -81,3 +82,24 @@ def get(self, request, url_slug):
             return Response(serializer.data, status=status.HTTP_200_OK)
         else:
             return Response(None, status=status.HTTP_404_NOT_FOUND)
+
+class HubThemeAPIView(APIView):
+    def get(self, request, url_slug):
+        try:
+            hub = Hub.objects.get(url_slug=str(url_slug))
+        except Hub.DoesNotExist:
+            return Response(
+                {"message": "Hub not found: {}".format(url_slug)},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        try:
+            hub_theme = HubTheme.objects.get(hub=hub)
+        except HubTheme.DoesNotExist:
+            return Response(
+                {"message": "Hub theme not found: {}".format(url_slug)},
+                status=status.HTTP_404_NOT_FOUND,
+            )
```

Comment:
```suggestion
            # returning an nearly empty object indicating no custom theme data for a valid hub without customization.
            return Response(
                {"message": "Hub theme not found: {}".format(url_slug)},
                status=status.HTTP_200_NOT_FOUND,
            )
```
I would prefer to send a 200 back, so that in the frontend we can distinguish an faulty request (hub not found) from a no custom data :)

Reply:
Imo since there is no hub theme found, 404 seems more reasonable because we are trying to retrieve the hub theme. Either way it can be distinguished by the error message. I'd vote to keeping the 404